collated/main/A0139257X.md

```diff
@@ -0,0 +1,1439 @@
+    public static final int MARCH = 3;
+    public static final int APRIL = 4;
+    public static final int MAY = 5;
+    public static final int JUNE = 6;
+    public static final int JULY = 7;
+    public static final int AUGUST = 8;
+    public static final int SEPTEMBER = 9;
+    public static final int OCTOBER = 10;
+    public static final int NOVEMBER = 11;
+    public static final int DECEMBER = 12;
+
+    public static final int MONDAY = 1;
+    public static final int TUESDAY = 2;
+    public static final int WEDNESDAY = 3;
+    public static final int THURSDAY = 4;
+    public static final int FRIDAY = 5;
+    public static final int SATURDAY = 6;
+    public static final int SUNDAY = 7;
+
+    public static final int NUM_DAYS_IN_A_WEEK = 7;
+    public static final int NUM_MONTHS_IN_A_YEAR = 12;
+    
+    public static final int NOT_A_VALID_MONTH = 0;
+    public static final int NOT_A_VALID_DAY_OF_THE_WEEK = 0;
+
+    public static final int FIRST_DAY_OF_THE_MONTH = 1;
+    
+    public static final String DATE_DELIMITER = " .-/";
+    
+    public static final String DEFAULT_DATE = getDefaultDate();
+
+    public static final Pattern TASK_DATE_ARGS_FORMAT = Pattern
+            .compile("(?<day>(3[0-1]|2[0-9]|1[0-9]|[1-9]))" + "(-)(?<month>(1[0-2]|[1-9]))" + "(-)(?<year>([0-9]{4}))");
+    private static final DateTimeFormatter dtf = DateTimeFormatter.ofPattern("EEEE, d MMMM yyyy");
+    private static final DateTimeFormatter standardFormat = DateTimeFormatter.ofPattern("d-MM-yyyy");
+    SimpleDateFormat sdf = new SimpleDateFormat("d M yyyy");
+    
+    public static final String MESSAGE_TASK_DATE_CONSTRAINTS =
+            "Task dates should be separated by '-' or '.' or '/'"
+            + "\nSpelling of month should be in full or 3-letters"
+            + "\nYear should only be 4-digits";
+
+    public String taskDate;
+    
+    /**
+     * Initialize the different fields given taskDate in the format of 
+     * DAY-MONTH-YEAR, separated by DATE_DELIMITER
+     * @throws IllegalValueException 
+     */
+    public TaskDate(String dateToAdd) throws IllegalValueException {
+        if (isValidFullDate(dateToAdd)) {
+            setDateGivenFullDate(dateToAdd);
+        } else if (isValidDayAndMonth(dateToAdd)) {
+            setDateGivenDayMonth(dateToAdd);
+        } else if (isValidMonthAndYear(dateToAdd)) {
+            setDateGivenMonthYear(dateToAdd);
+        } else if (isValidMonth(dateToAdd)) {
+            setDateGivenMonth(dateToAdd);
+        } else if (isValidDayOfWeek(dateToAdd)) {
+            setDateGivenDayNameOfWeek(dateToAdd);
+        } else if (isValidToday(dateToAdd)) {
+            setDateGivenToday(dateToAdd);
+        } else if (isValidTomorrow(dateToAdd)) {
+            setDateGivenTomorrow(dateToAdd);
+        } else {
+            throw new IllegalValueException(MESSAGE_TASK_DATE_CONSTRAINTS);
+        }
+    }
+    
+    /**
+     * Extract the different fields from taskDate having the format of
```

Comment:
Extract_**s**_ ...
similarly fix all other header comments to follow the accepted style